### DISCUSSION TITLE
Encoding of `explorerURLState`

### DISCUSSION THREAD

**Discussion by carloschida** [apollographql/apollo-studio-community#127] [Q&A] [answered]
Hi,

We are aiming to use Apollo Studio as a temporary substitute for our admin panel.
I've noticed that when I share a query/mutation, the link has two search params: `endpoint` and `explorerURLState`.

I believe the latter is the one that contains the information about the query in the UI editor. It would be really helpful to create this encoding programatically so that we can provide our admins with the right URL on each notification that requires their manual input.

The question is then, how os `explorerURLState` encoded and how can I create my own?

Thanks!

**Comment by cheapsteak** [comment, marked as answer]
Hi there, 

We use lz-string - https://github.com/pieroxy/lz-string 

Hope this sample would help!

```ts
import LZString from 'lz-string';

const explorerURLState = LZString.compressToEncodedURIComponent(
  JSON.stringify({ document, variables, headers, includeCookies }),
)
```

**Reply by dchambers** [reply]
Thank you kindly @cheapsteak. Here's a StackBlitz which demonstrates this:

- https://stackblitz.com/edit/apollo-operation-links?file=index.js

`npm test` to test.

**Reply by carloschida** [reply]
Sorry for the late reply. I muted my notifications. This is great to know. Thanks a lot.
